sw/device/boot_rom/boot_rom.c

```diff
@@ -12,21 +12,10 @@
 #include "sw/device/lib/spi_device.h"
 #include "sw/device/lib/uart.h"
 
-static noreturn void try_launch(void) {
-  asm volatile(
-      "  la   sp, _stack_start;"
-      "  tail _flash_start;"
-      // Unfortunately, we can't tell the compiler that we're about to clobber
-      // the stack pointer (which is meaningless, since there's nowhere to spill
-      // sp to). However, this doesn't matter, since this asm block will never
-      // return.
-      :
-      :
-      : /* sp, */ "memory");
-  __builtin_unreachable();
-}
+// NOTE: This symbol is generated by the linker. See rom_link.ld.
+extern noreturn void _flash_start(void);
 
-void _boot_start() {
+void _boot_start(void) {
```

Comment:
should this be marked `noreturn` too?

Reply:
No, it has an early return halfway through. I imagine that `_boot_start()` could return in the event of failing to jump to flash.